crates/sui-types/src/storage.rs

```diff
@@ -969,3 +969,138 @@ impl Display for DeleteKind {
         }
     }
 }
+
+#[derive(Clone, Debug, Default)]
+pub struct OneShotSharedInMemoryStore(SharedInMemoryStore);
```

Comment:
nit: maybe `SingleCheckpointInMemoryStore`? and a small comment why do we need one